config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -84,15 +99,24 @@
                                 final Element bundleElement = bundleDocumentBuilder.build(document, entities);
 
                                 // Create DELETE bundle - ALWAYS skip environment entities
-                                final Element deleteBundleElement = createDeleteBundle(document, entities, bundle,
-                                        annotatedEntity);
+                                Element deleteBundleElement = null;
+                                if (EntityBuilder.BundleType.DEPLOYMENT.equals(bundleType)) {
+                                    deleteBundleElement = createDeleteBundle(document, entities, bundle,
+                                            annotatedEntity);
+                                }
+
+                                // Create DELETE Environment bundle
+                                Element deleteEnvBundleElement = null;
+                                if (EntityBuilder.BundleType.ENVIRONMENT.equals(bundleType)) {
+                                    deleteEnvBundleElement = createDeleteEnvBundle(document, entities);
+                                }
 
```

Comment:
@uppoju , above logic is works good for environment delete bundle, but when we execute build-full-bundle, it should not create two separate delete bundles for policies and environment, it should be only one full delete bundle contains all entities with delete action

Reply:
full bundle delete bundle is handled in another PR 235